examples/inference/serving/torch_serve/Colossal_Inference_Handler.py

```diff
@@ -0,0 +1,210 @@
+from abc import ABC
+
+import torch
+import transformers
+from transformers import AutoTokenizer, BloomForCausalLM, BloomTokenizerFast, LlamaForCausalLM
+from ts.torch_handler.base_handler import BaseHandler
+
+import colossalai
+from colossalai.inference.tensor_parallel.engine import TPInferEngine
+from colossalai.shardformer import ShardConfig
+
+logger = logging.getLogger(__name__)
+logger.info("Transformers version %s", transformers.__version__)
+logger.info("ColossalAI version %s", colossalai.__version__)
+
+
+# from colossalai.testing
+# assins a random port, for demo use only
+def free_port() -> int:
+    """Get a free port on localhost.
+
+    Returns:
+        int: A free port on localhost.
+    """
+    while True:
+        port = random.randint(20000, 65000)
+        try:
+            with socket.socket() as sock:
+                sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                sock.bind(("localhost", port))
+                return port
+        except OSError:
+            continue
+
+
+class ColossalInferenceHandler(BaseHandler, ABC):
+    """
+    Transformers handler class for testing
+    """
+
+    def __init__(self):
+        super(ColossalInferenceHandler, self).__init__()
+        self.infer_engine = None
+        self.max_batch_size = None
+        self.max_input_len = None
+        self.max_output_len = None
+        self.tokenizer = None
+        self.initialized = False
+
+    def initialize(self, ctx):
+        """Expected behaviour: the sharded Bloom/Llama model is loaded.
+
+        Args:
+            ctx (context): It is a JSON Object containing information
+            pertaining to the model artefacts parameters.
+        """
+        if ctx is not None or not hasattr(ctx, "model_yaml_config"):
+            logger.error("Context ctx and model-config are not appropriately passed in.")
+
+        self.manifest = ctx.manifest
+        gpu_id = ctx.system_properties.get("gpu_id", -1)
+        model_dir = ctx.system_properties.get("model_dir")
+
+        # Inference configs are collected together in model yaml config for handler use
+        inference_config = ctx.model_yaml_config["handler"]
+        logger.info(inference_config)
+        inference_config["model_type"]
+        self.tp_size = inference_config.get("tp_size", 1)
+        self.max_batch_size = inference_config.get("max_batch_size", 4)
+        self.max_input_len = inference_config.get("max_input_len", 1024)
+        self.max_output_len = inference_config.get("max_output_len", 128)
+
+        self.device = torch.device("cuda:" + str(gpu_id) if torch.cuda.is_available() and gpu_id >= 0 else "cpu")
+        logger.info(f"Device set to {self.device}")
+        logger.info(f"torch.cuda.device_count() {torch.cuda.device_count()}")
+
+        logger.info(f"Unpacking from model_dir {model_dir}")
+        model_dir_path = os.path.join(model_dir, "model")
+        with zipfile.ZipFile(model_dir + "/model.zip", "r") as zip_ref:
```

Comment:
What's the format of model?

Reply:
A regular model repo or a snapshot on HuggingFace, which contains files like pytorch_modelXX.bin, config.json, tokenizer.json, etc